### ISSUE TITLE
Corrupt DB?

### ISSUE THREAD

**Issue by malx122** [Starcounter/Starcounter#3545]
I started to get failed startups with messages like this on the local machine:
```
ScErrUnexpectedCommandException (SCERR10016): Attempting to restart code host for default failed due to an unexpected problem.
Version: 2.2.1872.
Help page: https://github.com/Starcounter/Starcounter/wiki/SCERR10016.
 System.NullReferenceException: Object reference not set to an instance of an object.
   at Starcounter.Server.DatabaseEngine.RestartCodeHost(ICommandProcessor processor, Int32 terminatingCodeHostProcessId, DatabaseInfo databaseInfo) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-289\Level1\src\Server\Starcounter.Server\CoreServices\DatabaseEngine.cs:line 503
   at Starcounter.Server.Commands.CommandProcessor.ProcessCommand(NotifyCommandStatusChangedCallback notifyStatusChangedCallback) in C:\TeamCity\TeamCity9\buildAgent\work\sc-develop-nightly-289\Level1\src\Server\Starcounter.Server\Commands\CommandProcessor.cs:line 298
HResult=-2147467261
```
And then when restarting after `staradmin kill all` I get this all the time:
```
"Access violation (C0000005) detected at ..\..\src\sccoredb\sccoredb\sccoredb.cpp:452 (Thread 3964).
    0000022F54621653 sccoredb!get_slot_read_ptr : getslotreadptr.hpp (35)
    0000022F5461B47E sccoredb!__get_table_info : schemaapi.cpp (660)
    0000022F5460EAF2 sccoredb!sccoredb_get_table_info_by_name : sccoredb.cpp (443)
    00007FFC7A9EEEE4 <unknown>!
    0000022C11BBBB8C <unknown>!
    0000003D2397F600 <unknown>!
    0000022F54720000 <unknown>!
    0000022F54720940 <unknown>!
    00007F29B9852ECE <unknown>!
    00007FFCDA6F9598 clr!
    FFFFFFFFFFFFFFFF <unknown>!
    Endless stackwalk detected.
    An error occured during callstack dump.

```

**Comment by dan31** [comment]
cc @eriohl @k-rus

**Comment by malx122** [comment]
2.2.1872

**Comment by eriohl** [comment]
Certainly looks to be something wrong with the database structure. We recently fixed an issue related to metadata that messed up the database but I can't see if this is related. Anyway: If you could make the database available to us we'll look into this promptly (we need the image files and the transaction log)?

**Comment by malx122** [comment]
I have them stored but do not want to upload them public. Contact me via chat and I will give you a link

**Comment by eriohl** [comment]
Chat? Like gitter? Is this a production database?

**Comment by malx122** [comment]
Google hangout? Skype? Yea, I had a copy of a production database locally to test things. (However it has worked for a long time before this happened so it was not a problem in the copy/move process)

**Comment by eriohl** [comment]
Confirmed that this is the same bug that was detected and addressed earlier this week (#3532). Thank you for your help @malx122.
